app/controllers/pxe_controller/pxe_image_types.rb

```diff
@@ -12,65 +12,10 @@ def pxe_image_type_new
 
   def pxe_image_type_edit
     assert_privileges("pxe_image_type_edit")
-    if params[:button] == "cancel"
-      id = params[:id] || "new"
-      return unless load_edit("pxe_image_type_edit__#{id}", "replace_cell__explorer")
-      if @edit[:pxe_id]
-        add_flash(_("Edit of System Image Type \"%{name}\" was cancelled by the user") % {:name => @edit[:current][:name]})
-      else
-        add_flash(_("Add of new System Image Type was cancelled by the user"))
-      end
-      @edit = session[:edit] = nil # clean out the saved info
-      get_node_info(x_node)
-      replace_right_cell(:nodetype => x_node)
-    elsif %w[add save].include?(params[:button])
-      id = params[:id] || "new"
-      return unless load_edit("pxe_image_type_edit__#{id}", "replace_cell__explorer")
-      pxe_image_type_get_form_vars
-      add_pxe = params[:id] ? find_record_with_rbac(PxeImageType, params[:id]) : PxeImageType.new
-      pxe_image_type_validate_fields
-      if @flash_array
-        javascript_flash
-        return
-      end
-
-      pxe_image_type_set_record_vars(add_pxe)
-
-      if add_pxe.save
-        AuditEvent.success(build_created_audit(add_pxe, @edit))
-        @edit = session[:edit] = nil # clean out the saved info
-        add_flash(_("System Image Type \"%{name}\" was added") % {:name => add_pxe.name})
-        get_node_info(x_node)
-        replace_right_cell(:nodetype => x_node, :replace_trees => %i[pxe_image_types customization_templates])
-      else
-        @in_a_form = true
-        add_pxe.errors.each do |field, msg|
-          add_flash("#{field.to_s.capitalize} #{msg}", :error)
-        end
-        javascript_flash
-      end
-    else
-      # first time in or reset
-      add_flash(_("All changes have been reset"), :warning) if params[:button] == "reset"
-      @pxe_image_type = @record = find_record_with_rbac(PxeImageType, @_params[:id] = checked_or_params.first)
-      pxe_image_type_set_form_vars
-      @in_a_form = true
-      session[:changed] = false
-      replace_right_cell(:nodetype => "pit")
-    end
-  end
-
-  # AJAX driven routine to check for changes in ANY field on the form
-  def pxe_image_type_form_field_changed
-    assert_privileges(params[:id] == 'new' ? 'pxe_image_type_new' : 'pxe_image_type_edit')
-    return unless load_edit("pxe_image_type_edit__#{params[:id]}", "replace_cell__explorer")
-    pxe_image_type_get_form_vars
-    render :update do |page|
-      page << javascript_prologue
-      page.replace_html("form_div", :partial => "pxe_image_type_form") if params[:provision_type]
-      changed = (@edit[:new] != @edit[:current])
-      page << javascript_for_miq_button_visibility(changed)
-    end
+    @pxe_image_type = @record = find_record_with_rbac(PxeImageType, @_params[:id] = checked_or_params.first)
+    pxe_image_type_set_form_vars
```

Comment:
Removing the `pxe_image_type_set_form_vars` method breaks both the new and editing forms so I left it in for now

Reply:
Breaks it how?

Reply:
![image](https://user-images.githubusercontent.com/64800041/106619328-7a599100-653e-11eb-9909-b5450afcf525.png)

Reply:
```diff
diff --git a/app/controllers/pxe_controller.rb b/app/controllers/pxe_controller.rb
index d4eb1d0c72..3c0f335a02 100644
--- a/app/controllers/pxe_controller.rb
+++ b/app/controllers/pxe_controller.rb
@@ -251,7 +251,7 @@ def replace_right_cell(options = {})
         presenter.update(:form_buttons_div, r[
           :partial => "layouts/x_edit_buttons",
           :locals  => {
-            :record_id    => @edit[:rec_id],
+            :record_id    => @edit.try(:[], :rec_id),
             :action_url   => action_url,
             :multi_record => multi_record,
             :serialize    => true
diff --git a/app/controllers/pxe_controller/pxe_image_types.rb b/app/controllers/pxe_controller/pxe_image_types.rb
index f69dba04d3..e745d44139 100644
--- a/app/controllers/pxe_controller/pxe_image_types.rb
+++ b/app/controllers/pxe_controller/pxe_image_types.rb
@@ -5,7 +5,6 @@ module PxeController::PxeImageTypes
   def pxe_image_type_new
     assert_privileges("pxe_image_type_new")
     @pxe_image_type = PxeImageType.new
-    pxe_image_type_set_form_vars
     @in_a_form = true
     replace_right_cell(:nodetype => "pit")
   end
@@ -13,7 +12,6 @@ def pxe_image_type_new
   def pxe_image_type_edit
     assert_privileges("pxe_image_type_edit")
     @pxe_image_type = @record = find_record_with_rbac(PxeImageType, @_params[:id] = checked_or_params.first)
-    pxe_image_type_set_form_vars
     @in_a_form = true
     replace_right_cell(:nodetype => "pit")
   end
diff --git a/app/views/pxe/_pxe_image_type_details.html.haml b/app/views/pxe/_pxe_image_type_details.html.haml
index 64f85dcca3..6b212dd91a 100644
--- a/app/views/pxe/_pxe_image_type_details.html.haml
+++ b/app/views/pxe/_pxe_image_type_details.html.haml
@@ -1,7 +1,7 @@
 #pxe_image_type_details_div
   -# if PXE Server is selected %>
   - if @pxe_image_type
-    - if @edit
+    - if @in_a_form
       = render :partial => "pxe_image_type_form"
 
     - else
```